src/Approximations/iterative_refinement.jl

```diff
@@ -95,43 +158,44 @@ local `Approximation2D`.
 
 An ɛ-close approximation of the given 2D convex set.
 """
-function approximate(S::LazySet, ɛ::Float64)::Vector{Approximation2D}
+function approximate(S::LazySet, ɛ::N)::PolygonalOverapproximation{N} where {N<:AbstractFloat}
 
-    # start with box directions
+    # initialize box directions
     pe = σ(DIR_EAST, S)
     pn = σ(DIR_NORTH, S)
     pw = σ(DIR_WEST, S)
     ps = σ(DIR_SOUTH, S)
-    queue = Approximation2D[]
-    push!(queue, Approximation2D(pe, DIR_EAST, pn, DIR_NORTH))
-    push!(queue, Approximation2D(pn, DIR_NORTH, pw, DIR_WEST))
-    push!(queue, Approximation2D(pw, DIR_WEST, ps, DIR_SOUTH))
-    push!(queue, Approximation2D(ps, DIR_SOUTH, pe, DIR_EAST))
+
+    Ω = PolygonalOverapproximation{N}(S)
+
+    addapproximation!(Ω, pe, DIR_EAST, pn, DIR_NORTH)
+    addapproximation!(Ω, pn, DIR_NORTH, pw, DIR_WEST)
+    addapproximation!(Ω, pw, DIR_WEST, ps, DIR_SOUTH)
+    addapproximation!(Ω, ps, DIR_SOUTH, pe, DIR_EAST)
 
     i = 1
-    N = length(queue)
-    while i <= length(queue)
-        if (queue[i].err <= ɛ)
+    while i <= length(Ω.approx_list)
+        if Ω.approx_list[i].err <= ɛ || !Ω.approx_list[i].refinable
+            # if this approximation doesn't need to be refined, consider the next
+            # one in the queue (counter-clockwise order wrt d1)
+            # if the approximation is not refinable => continue
```

Comment:
The two conditions sound almost the same to me - is there a typo?

Reply:
no typo. but we don't need it anymore, and there is anyways the assertion `@assert R.refinable` inside the function`refinable` => keep `Ω.approx_list[i].err <= ɛ`   👍